Library/Homebrew/download_strategy.rb

```diff
@@ -388,6 +388,25 @@ def _fetch
   end
 end
 
+class CurlSourceForgeSecureMirrorDownloadStrategy < CurlDownloadStrategy
+  def fetch
+    if !ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+      oldurl = @url
+      @url = https_mirror
+      ohai "Sourceforge HTTPS: #{oldurl}"
+      ohai "=> converted to #{@url}"
+    end
+    super
+  end
+
+  def https_mirror
+     re = /(^https?:\/\/)(.+\.net\/project\/)(.+)/
```

Comment:
You can use `%r{...}` instead of `/.../` to avoid having to escape all slashes (e.g. `%r{/}` instead of `/\//`).